# ESAA 팀 과제 : 유튜브 데이터 분석
## 2019년 10월 11일 
### 김지원, 박보정, 유찬미, 예지혜

### 0. 영국.미국.캐나다의 자료를 모두 통합하여 조회수로 정렬한 뒤, 어떤 나라가 가장 조회수가 많은지 알아보겠다. 

1) 파일불러오기 

In [5]:
import pandas as pd
us = pd.read_csv('C:/temp/yt/USvideos.csv')
ca = pd.read_csv('C:/temp/yt/CAvideos.csv')
gb = pd.read_csv('C:/temp/yt/GBvideos.csv')

country 변수 추가하기 

In [16]:
us["country"] = "US"
ca["country"] = "CA"
gb["country"] = "GB"

2) merge실행 

In [15]:
list(us)==list(ca)
list(us)==list(us)

True

In [19]:
print(us.shape)
print(ca.shape)
print(gb.shape)

(40949, 17)
(40881, 17)
(38916, 17)


In [24]:
youtube_merge = pd.concat([ca,us,gb])
print(youtube_merge.shape)

(120746, 17)


3) 조회수로 sort 

4) 100위까지 자르기. 

In [33]:
youtube_sort = youtube_merge.sort_values(["views"], ascending=[False])
youtube_sort=youtube_sort.reset_index(drop=True)

In [35]:
youtube_100 = youtube_sort.iloc[0:100, ]
youtube_100

video_id trending_date  \
0   _I_D_8Z4sJE      18.07.04   
1   _I_D_8Z4sJE      18.06.04   
2   _I_D_8Z4sJE      18.05.04   
3   _I_D_8Z4sJE      18.04.04   
4   _I_D_8Z4sJE      18.03.04   
5   _I_D_8Z4sJE      18.02.04   
6   _I_D_8Z4sJE      18.01.04   
7   _I_D_8Z4sJE      18.31.03   
8   _I_D_8Z4sJE      18.30.03   
9   9jI-z9QN6g8      18.18.05   
10  kLpH1nSLJSs      18.23.03   
11  9jI-z9QN6g8      18.17.05   
12  _I_D_8Z4sJE      18.29.03   
13  kLpH1nSLJSs      18.22.03   
14  kLpH1nSLJSs      18.21.03   
15  _I_D_8Z4sJE      18.28.03   
16  kLpH1nSLJSs      18.20.03   
17  kLpH1nSLJSs      18.19.03   
18  kLpH1nSLJSs      18.18.03   
19  _I_D_8Z4sJE      18.27.03   
20  9jI-z9QN6g8      18.14.05   
21  kLpH1nSLJSs      18.17.03   
22  9jI-z9QN6g8      18.13.05   
23  kLpH1nSLJSs      18.16.03   
24  wfWkmURBNv8      18.01.03   
25  _I_D_8Z4sJE      18.26.03   
26  kLpH1nSLJSs      18.15.03   
27  wfWkmURBNv8      18.28.02   
28  kLpH1nSLJSs      18.14.03   
29  9jI-z9QN6g8      18.12.05   
..          ...           ...   
70  VYOjWnS4cMY      18.01.06   
71  xpVfcZ0ZcFM      18.20.03   
72  wfWkmURBNv8      18.22.02   
73  VYOjWnS4cMY      18.31.05   
74  VYOjWnS4cMY      18.31.05   
75  xpVfcZ0ZcFM      18.19.03   
76  kLpH1nSLJSs      18.06.03   
77  _I_D_8Z4sJE      18.19.03   
78  9jI-z9QN6g8      18.06.05   
79  VYOjWnS4cMY      18.30.05   
80  VYOjWnS4cMY      18.30.05   
81  wfWkmURBNv8      18.21.02   
82  ffxKSjUwKdU      18.26.05   
83  VYOjWnS4cMY      18.29.05   
84  VYOjWnS4cMY      18.29.05   
85  xpVfcZ0ZcFM      18.18.03   
86  kLpH1nSLJSs      18.05.03   
87  ffxKSjUwKdU      18.25.05   
88  _I_D_8Z4sJE      18.18.03   
89  zEf423kYfqk      18.26.05   
90  VYOjWnS4cMY      18.28.05   
91  VYOjWnS4cMY      18.28.05   
92  ffxKSjUwKdU      18.24.05   
93  wfWkmURBNv8      18.20.02   
94  VYOjWnS4cMY      18.27.05   
95  VYOjWnS4cMY      18.27.05   
96  xpVfcZ0ZcFM      18.17.03   
97  9jI-z9QN6g8      18.05.05   
98  ffxKSjUwKdU      18.23.05   
99  zEf423kYfqk      18.25.05   

                                                title        channel_title  \
0   Nicky Jam x J. Balvin - X (EQUIS) | Video Ofic...           NickyJamTV   
1   Nicky Jam x J. Balvin - X (EQUIS) | Video Ofic...           NickyJamTV   
2   Nicky Jam x J. Balvin - X (EQUIS) | Video Ofic...           NickyJamTV   
3   Nicky Jam x J. Balvin - X (EQUIS) | Video Ofic...           NickyJamTV   
4   Nicky Jam x J. Balvin - X (EQUIS) | Video Ofic...           NickyJamTV   
5   Nicky Jam x J. Balvin - X (EQUIS) | Video Ofic...           NickyJamTV   
6   Nicky Jam x J. Balvin - X (EQUIS) | Video Ofic...           NickyJamTV   
7   Nicky Jam x J. Balvin - X (EQUIS) | Video Ofic...           NickyJamTV   
8   Nicky Jam x J. Balvin - X (EQUIS) | Video Ofic...           NickyJamTV   
9   Te Bote Remix - Casper, Nio García, Darell, Ni...        Flow La Movie   
10               Bad Bunny - Amorfoda | Video Oficial            Bad Bunny   
11  Te Bote Remix - Casper, Nio García, Darell, Ni...        Flow La Movie   
12  Nicky Jam x J. Balvin - X (EQUIS) | Video Ofic...           NickyJamTV   
13               Bad Bunny - Amorfoda | Video Oficial            Bad Bunny   
14               Bad Bunny - Amorfoda | Video Oficial            Bad Bunny   
15  Nicky Jam x J. Balvin - X (EQUIS) | Video Ofic...           NickyJamTV   
16               Bad Bunny - Amorfoda | Video Oficial            Bad Bunny   
17               Bad Bunny - Amorfoda | Video Oficial            Bad Bunny   
18               Bad Bunny - Amorfoda | Video Oficial            Bad Bunny   
19  Nicky Jam x J. Balvin - X (EQUIS) | Video Ofic...           NickyJamTV   
20  Te Bote Remix - Casper, Nio García, Darell, Ni...        Flow La Movie   
21               Bad Bunny - Amorfoda | Video Oficial            Bad Bunny   
22  Te Bote Remix - Casper, Nio García, Darell, Ni...        Flow La Movie   
23               Bad Bunny - Amorfoda | Video Oficial            Bad Bunny   
24        

5) 나라별로 갯수세기

In [40]:
youtube_100["country"].value_counts()

GB    93
US     7
Name: country, dtype: int64

In [42]:
youtube_100["country"].head()

0    GB
1    GB
2    GB
3    GB
4    GB
Name: country, dtype: object

#### < 결론 >
#### 캐나다, 미국, 영국의 자료들을 합쳐서 조회수로 나열 후, 100위까지 확인해본 결과 영국의 동영상이 93개, 미국의 동영상이 7개가 있었다. 맨 앞의 5위까지 모두 영국의 자료로 확인할 수 있다. 세 나라 중 영국의 유튜브가 조회수가 가장 많음을 알 수 있다. 

### 1. 조회수 100위까지 자르고, 태그 수를 비교/어떤 태그가 많이 사용되었는지 알아보기

1) 파일 불러오기

In [2]:
import pandas as pd
input_fd = open('KRvideos.csv',errors = 'backslashreplace',encoding='utf-8')
kr = pd.read_csv(input_fd)

In [16]:
print(kr.shape)

(34567, 16)


2) 조회수 정렬 후 100위까지 자르기

In [56]:
kr100 = kr.sort_values('views', ascending=False).iloc[:100,:]

In [54]:
kr100.head()

video_id trending_date  \
4856   FlsCjmMhFmw      17.11.12   
4678   FlsCjmMhFmw      17.10.12   
4018   6ZfuNTqbHE8      17.06.12   
30844  7C2z4GqqS5E      18.24.05   
3750   6ZfuNTqbHE8      17.05.12   

                                                   title  \
4856   YouTube Rewind: The Shape of 2017 | #YouTubeRe...   
4678   YouTube Rewind: The Shape of 2017 | #YouTubeRe...   
4018   Marvel Studios' Avengers: Infinity War Officia...   
30844                BTS (방탄소년단) 'FAKE LOVE' Official MV   
3750   Marvel Studios' Avengers: Infinity War Officia...   

              channel_title  category_id              publish_time  \
4856      YouTube Spotlight           24  2017-12-06T17:58:51.000Z   
4678      YouTube Spotlight           24  2017-12-06T17:58:51.000Z   
4018   Marvel Entertainment           24  2017-11-29T13:26:24.000Z   
30844               ibighit           10  2018-05-18T09:00:02.000Z   
3750   Marvel Entertainment           24  2017-11-29T13:26:24.000Z   

                                                    tags      views    likes  \
4856   Rewind|"Rewind 2017"|"youtube rewind 2017"|"#Y...  113876217  2811223   
4678   Rewind|"Rewind 2017"|"youtube rewind 2017"|"#Y...  100911567  2656686   
4018   marvel|"comics"|"comic books"|"nerdy"|"geeky"|...   87450245  2584675   
30844  BIGHIT|"빅히트"|"방탄소년단"|"BTS"|"BANGTAN"|"방탄"|"FAK...   87430865  5150839   
3750   marvel|"comics"|"comic books"|"nerdy"|"geeky"|...   84281319  2555414   

       dislikes  comment_count  \
4856    1470388         787174   
4678    1353663         682890   
4018      52176         341571   
30844    172570        1142274   
3750      51008         339708   

                                       thumbnail_link  comments_disabled  \
4856   https://i.ytimg.com/vi/FlsCjmMhFmw/default.jpg              False   
4678   https://i.ytimg.com/vi/FlsCjmMhFmw/default.jpg              False   
4018   https://i.ytimg.com/vi/6ZfuNTqbHE8/default.jpg              False   
30844  https://i.ytimg.com/vi/7C2z4GqqS5E/default.jpg              False   
3750   https://i.ytimg.com/vi/6ZfuNTqbHE8/default.jpg              False   

       ratings_disabled  video_error_or_removed  \
4856              False                   False   
4678              False                   False   
4018              False                   False   
30844             False                   False   
3750              False                   False   

                                             description  
4856   YouTube Rewind 2017. Celebrating the videos, p...  
4678   YouTube Rewind 2017. Celebrating the videos, p...  
4018   There was an idea… Avengers: Infinity War. In ...  
30844  BTS (방탄소년단) 'FAKE LOVE' Official MVDirector : ...  
3750   There was an idea… Avengers: Infinity War. In ...

In [57]:
kr100.shape

(100, 16)

3) 태그 수를 집계한 tag_count 컬럼 추가하기

In [58]:
kr100['tag_count'] = kr100['tags'].apply(lambda x:x.count('|')+1)

In [59]:
print(kr100.iloc[1].tags)
print(kr100.iloc[1].tag_count)

Rewind|"Rewind 2017"|"youtube rewind 2017"|"#YouTubeRewind"|"Rewind 2016"|"Dan and Phil"|"Grace Helbig"|"HolaSoyGerman"|"Lilly Singh"|"Markiplier"|"Swoozie"|"Rhett Link"|"Liza Koshy"|"Dolan Twins"|"Lele Pons"|"Jake Paul"|"Logan Paul"|"KSI"|"Joey Graceffa"|"Casey Neistat"|"Poppy"|"Niana Guerrero"|"Daddy Yankee"|"Luis Fonsi"|"Ed Sheeran"|"Kendrick Lamar"|"Stephen Colbert"|"Fidget Spinners"|"Slime"|"Backpack Kid"|"April the Giraffe"|"#Rewind"|"Despacito"|"Shape of you"|"YouTubeRewind"|"I’m the One"|"Humble"
37


#### 1-1. 태그 수와 조회수의 상관관계 분석하기 

In [60]:
kr100['ranking_group']=sorted(list(range(0,10))*10)

In [64]:
print(kr100.groupby('ranking_group').tag_count.mean())

ranking_group
0    18.1
1     8.1
2    23.0
3    17.3
4    14.1
5    13.2
6    20.0
7    19.9
8    15.4
9    17.3
Name: tag_count, dtype: float64


조회수 상위 100개 동영상에서 10개씩 동영상을 그룹화하여 태그 수를 비교하였을 때, 태그 수에 따른 조회수 차이는 없는 것으로 보인다.

#### 1-2. 조회수 상위 100개 동영상에서 많이 사용된 태그 찾기

In [80]:
def string_handle(x): #문자열 처리 함수
    x = x.lower() #소문자로 변경
    x = x.replace('"','') #"제거
    x = x.replace('#','') ##제거
    x = x.replace('_', ' ') #_를 공백으로 변경
    x = x.strip() #양쪽 공백 제거
    return x

In [74]:
tag100 = []
for tags in kr100['tags']:
    tag_lst = tags.split('|')
    tag100.extend(tag_lst)

In [81]:
handled_tag100 = [string_handle(x) for x in tag100]

In [119]:
tag100_count=pd.DataFrame(data={'tag':handled_tag100}).groupby('tag').size().reset_index().rename(columns={0:'count'}).sort_values('count',ascending=False)
tag100_count.head(15)

tag  count
250        marvel     26
360     superhero     18
149     fake love     18
359    super hero     18
72         comics     18
71    comic books     18
202  infinity war     18
26        bangtan     15
32         bighit     15
44            bts     15
20       avengers     15
452            방탄     15
453         방탄소년단     15
456           빅히트     15
290           pop     15

많이 사용된 태그 상위 15개에서 pop을 제외한 14개 모두 marvel이나 방탄소년단과 연관된 태그이다. 17~18년도 한국 유튜브 트랜드는 마블 영화인 어밴져스와 방탄소년단이 지배적이였던 것으로 보인다.

#### 1-3. 캐나다, 미국, 영국의 조회수 상위 100개 영상의 태그 트렌드 분석

In [122]:
ca=pd.read_csv('CAvideos.csv')
us=pd.read_csv('USvideos.csv')
gb=pd.read_csv('GBvideos.csv')

In [128]:
def top_tag10(data):
    df100 = data.sort_values('views', ascending=False).iloc[:100,:]
    tag100 = []
    for tags in df100['tags']:
        tag_lst = tags.split('|')
        tag100.extend(tag_lst)
    handled_tag100 = [string_handle(x) for x in tag100]
    tag100_count=pd.DataFrame(data={'tag':handled_tag100}).groupby('tag').size().reset_index().rename(columns={0:'count'}).sort_values('count',ascending=False)
    return tag100_count.iloc[:10,:]

In [136]:
top_tag10(ca)

tag  count
306            pop     21
264         marvel     19
387      superhero     14
386     super hero     14
85          comics     14
84     comic books     14
189   infinity war     14
27        avengers     14
126     ed sheeran     13
464  youtuberewind     12

캐나다도 한국과 마찬가지로 marvel의 어밴저스가 지배적인것을 볼 수 있다. 더불어 Ed Sheeran의 노래도 인기를 끈 것으로 보인다.

In [137]:
top_tag10(us)

tag  count
57                    fake love     34
137                         rap     29
112  mcdj recording/rca records     22
159             this is america     22
35             childish gambino     22
135                         pop     21
138                     records     16
15                          bts     12
38                          cry     12
131                          no     12

미국은 오히려 순위권에 marvel 관련 태그가 없고 비교적 태그들이 다양하게 분포하고 있다. 다른 나라의 유튜브 트렌드와 다른 점은 순위권에 있는 많은 태그가 대중음악와 관련되어 있다는 점이다. 방탄소년단의 fake love, Childish Gambino의 This is America가 인기를 끌었고, 그 밖에도 rap, pop, recoreds 등 대부분의 태그가 음악과 관련되어 있다.

In [138]:
top_tag10(gb)

tag  count
4           bad bunny     54
60              ozuna     45
1            amorfoda     42
69       romeo santos     33
26  el farsante remix     33
62            piso 21     22
68               reik     22
23               dura     22
24  déjala que vuelva     22
25         ed sheeran     22

영국도 미국과 마찬가지로 상위 100개 영상의 대부분의 태그가 대중음악과 관련된 태그였다. 영국에는 가수 bad bunny의 amorfoda라는 노래가 선풍적인 인기를 끈 것으로 보인다. 100개 영상 중 거의 절반이 이 곡과 관련된 태그를 가지고 있었다. 영국 유튜브에서 의외인 점은 태그 대부분이 스페인 노래와 관련되었다는 점이다.

### 2. 댓글 수 100위까지 자르고, 태그 수와 댓글 수의 상관관계 분석

In [3]:
kr100 = KRvideos.sort_values('comment_count', ascending=False).iloc[:100,:]

In [4]:
kr100.head()

video_id trending_date                                title  \
30844  7C2z4GqqS5E      18.24.05  BTS (방탄소년단) 'FAKE LOVE' Official MV   
30583  7C2z4GqqS5E      18.23.05  BTS (방탄소년단) 'FAKE LOVE' Official MV   
30352  7C2z4GqqS5E      18.22.05  BTS (방탄소년단) 'FAKE LOVE' Official MV   
30486  7C2z4GqqS5E      18.22.05  BTS (방탄소년단) 'FAKE LOVE' Official MV   
30291  7C2z4GqqS5E      18.21.05  BTS (방탄소년단) 'FAKE LOVE' Official MV   

      channel_title  category_id              publish_time  \
30844       ibighit           10  2018-05-18T09:00:02.000Z   
30583       ibighit           10  2018-05-18T09:00:02.000Z   
30352       ibighit           10  2018-05-18T09:00:02.000Z   
30486       ibighit           10  2018-05-18T09:00:02.000Z   
30291       ibighit           10  2018-05-18T09:00:02.000Z   

                                                    tags     views    likes  \
30844  BIGHIT|"빅히트"|"방탄소년단"|"BTS"|"BANGTAN"|"방탄"|"FAK...  87430865  5150839   
30583  BIGHIT|"빅히트"|"방탄소년단"|"BTS"|"BANGTAN"|"방탄"|"FAK...  80738011  5053338   
30352  BIGHIT|"빅히트"|"방탄소년단"|"BTS"|"BANGTAN"|"방탄"|"FAK...  73463137  4924056   
30486  BIGHIT|"빅히트"|"방탄소년단"|"BTS"|"BANGTAN"|"방탄"|"FAK...  73463137  4924056   
30291  BIGHIT|"빅히트"|"방탄소년단"|"BTS"|"BANGTAN"|"방탄"|"FAK...  65396157  4750254   

       dislikes  comment_count  \
30844    172570        1142274   
30583    165854        1114800   
30352    156026        1084435   
30486    156026        1084435   
30291    141966        1040912   

                                       thumbnail_link  comments_disabled  \
30844  https://i.ytimg.com/vi/7C2z4GqqS5E/default.jpg              False   
30583  https://i.ytimg.com/vi/7C2z4GqqS5E/default.jpg              False   
30352  https://i.ytimg.com/vi/7C2z4GqqS5E/default.jpg              False   
30486  https://i.ytimg.com/vi/7C2z4GqqS5E/default.jpg              False   
30291  https://i.ytimg.com/vi/7C2z4GqqS5E/default.jpg              False   

       ratings_disabled  video_error_or_removed  \
30844             False                   False   
30583             False                   False   
30352             False                   False   
30486             False                   False   
30291             False                   False   

                                             description  
30844  BTS (방탄소년단) 'FAKE LOVE' Official MVDirector : ...  
30583  BTS (방탄소년단) 'FAKE LOVE' Official MVDirector : ...  
30352  BTS (방탄소년단) 'FAKE LOVE' Official MVDirector : ...  
30486  BTS (방탄소년단) 'FAKE LOVE' Official MVDirector : ...  
30291  BTS (방탄소년단) 'FAKE LOVE' Official MVDirector : ...

In [5]:
kr100['tag_count'] = kr100['tags'].apply(lambda x:x.count('|')+1)

In [10]:
kr100[['title','tags','views','comment_count','tag_count']]

title  \
30844                BTS (방탄소년단) 'FAKE LOVE' Official MV   
30583                BTS (방탄소년단) 'FAKE LOVE' Official MV   
30352                BTS (방탄소년단) 'FAKE LOVE' Official MV   
30486                BTS (방탄소년단) 'FAKE LOVE' Official MV   
30291                BTS (방탄소년단) 'FAKE LOVE' Official MV   
30171                BTS (방탄소년단) 'FAKE LOVE' Official MV   
30069                BTS (방탄소년단) 'FAKE LOVE' Official MV   
29969                BTS (방탄소년단) 'FAKE LOVE' Official MV   
4453   YouTube Rewind: The Shape of 2017 | #YouTubeRe...   
4856   YouTube Rewind: The Shape of 2017 | #YouTubeRe...   
4244   YouTube Rewind: The Shape of 2017 | #YouTubeRe...   
4678   YouTube Rewind: The Shape of 2017 | #YouTubeRe...   
2788   BTS (방탄소년단) 'MIC Drop (Steve Aoki Remix)' Offi...   
2563   BTS (방탄소년단) 'MIC Drop (Steve Aoki Remix)' Offi...   
2330   BTS (방탄소년단) 'MIC Drop (Steve Aoki Remix)' Offi...   
2149   BTS (방탄소년단) 'MIC Drop (Steve Aoki Remix)' Offi...   
28803  CHẠY NGAY ĐI | RUN NOW | SƠN TÙNG M-TP | Offic...   
1976   BTS (방탄소년단) 'MIC Drop (Steve Aoki Remix)' Offi...   
17879                         j-hope 'Daydream (백일몽)' MV   
28605  BTS (방탄소년단) LOVE YOURSELF 轉 Tear 'Singularity'...   
28371  BTS (방탄소년단) LOVE YOURSELF 轉 Tear 'Singularity'...   
4018   Marvel Studios' Avengers: Infinity War Officia...   
28139  BTS (방탄소년단) LOVE YOURSELF 轉 Tear 'Singularity'...   
3750   Marvel Studios' Avengers: Infinity War Officia...   
3544   Marvel Studios' Avengers: Infinity War Officia...   
3381   Marvel Studios' Avengers: Infinity War Officia...   
27955  BTS (방탄소년단) LOVE YOURSELF 轉 Tear 'Singularity'...   
3185   Marvel Studios' Avengers: Infinity War Officia...   
3330   Marvel Studios' Avengers: Infinity War Officia...   
3018   Marvel Studios' Avengers: Infinity War Officia...   
...                                                  ...   
26391                      VENOM - Official Trailer (HD)   
19908                                      GOT7 Look M/V   
29881          BTS (방탄소년단) 'FAKE LOVE' Official Teaser 2   
26147                      VENOM - Official Trailer (HD)   
5053                              TWICE Heart Shaker M/V   
20239  Marvel Studios' Avengers: Infinity War - Offic...   
20386  Marvel Studios' Avengers: Infinity War - Offic...   
13442                       Red Velvet 레드벨벳 'Bad Boy' MV   
158           Eminem - Walk On Water (Audio) ft. Beyoncé   
10107     Taylor Swift - End Game ft. Ed Sheeran, Future   
29670          BTS (방탄소년단) 'FAKE LOVE' Official Teaser 2   
29902  Do You Hear Yanny or Laurel? (SOLVED with SCIE...   
19880                            Taylor Swift - Delicate   
13216                       Red Velvet 레드벨벳 'Bad Boy' MV   
13289                       Red Velvet 레드벨벳 'Bad Boy' MV   
25972                      VENOM - Official Trailer (HD)   
9446   Bruno Mars - Finesse (Remix) [Feat. Cardi B] [...   
13161                       Red Velvet 레드벨벳 'Bad Boy' MV   
13063                       Red Velvet 레드벨벳 'Bad Boy' MV   
29453          BTS (방탄소년단) 'FAKE LOVE' Official Teaser 2   
29592          BTS (방탄소년단) 'FAKE LOVE' Official Teaser 2   
11989                    JONGHYUN 종현 '빛이 나 (Shinin’)' MV   
9244   Bruno Mars - Finesse (Remix) [Feat. Cardi B] [...   
12880                       Red Velvet 레드벨벳 'Bad Boy' MV   
13015                       Red Velvet 레드벨벳 'Bad Boy' MV   
4885                              TWICE Heart Shaker M/V   
11770                    JONGHYUN 종현 '빛이 나 (Shinin’)' MV   
19736                                      GOT7 Look M/V   
16219                                 Drake - God’s Plan   
32570              Maroon 5 - Girls Like You ft. Cardi B   

                                                    tags      views  \
30844  BIGHIT|"빅히트"|"방탄소년단"|"BTS"|"BANGTAN"|"방탄"|"FAK...   87430865   
30583  BIGHIT|"빅히트"|"방탄소년단"|"BTS"|"BANGTAN"|"방탄"|"FAK...   80738011   
30352  BIGHIT|"빅히트"|"방탄소년단"|"BTS"|"BANGTAN"|"방탄"|"FAK...   73463137   
30486  BIGHIT|"빅히트"|"방탄소년단"|"BTS"|"BANGTAN"|"방

댓글수 100위까지의 태그들과 태그 수를 살펴보면 어떠한 상관관계가 보이지는 않는다.

In [11]:
kr100['ranking_group']=sorted(list(range(0,10))*10)
print(kr100.groupby('ranking_group').views.mean())

ranking_group
0    76132593.0
1    33169677.4
2    57183768.6
3    23961329.2
4    31088894.3
5    22008246.2
6    23386064.6
7    23620077.1
8    20165450.1
9    12979505.9
Name: views, dtype: float64


In [20]:
KR_sort=KRvideos.sort_values('comment_count', ascending=False)
KR_sort.head()

video_id trending_date                                title  \
30844  7C2z4GqqS5E      18.24.05  BTS (방탄소년단) 'FAKE LOVE' Official MV   
30583  7C2z4GqqS5E      18.23.05  BTS (방탄소년단) 'FAKE LOVE' Official MV   
30352  7C2z4GqqS5E      18.22.05  BTS (방탄소년단) 'FAKE LOVE' Official MV   
30486  7C2z4GqqS5E      18.22.05  BTS (방탄소년단) 'FAKE LOVE' Official MV   
30291  7C2z4GqqS5E      18.21.05  BTS (방탄소년단) 'FAKE LOVE' Official MV   

      channel_title  category_id              publish_time  \
30844       ibighit           10  2018-05-18T09:00:02.000Z   
30583       ibighit           10  2018-05-18T09:00:02.000Z   
30352       ibighit           10  2018-05-18T09:00:02.000Z   
30486       ibighit           10  2018-05-18T09:00:02.000Z   
30291       ibighit           10  2018-05-18T09:00:02.000Z   

                                                    tags     views    likes  \
30844  BIGHIT|"빅히트"|"방탄소년단"|"BTS"|"BANGTAN"|"방탄"|"FAK...  87430865  5150839   
30583  BIGHIT|"빅히트"|"방탄소년단"|"BTS"|"BANGTAN"|"방탄"|"FAK...  80738011  5053338   
30352  BIGHIT|"빅히트"|"방탄소년단"|"BTS"|"BANGTAN"|"방탄"|"FAK...  73463137  4924056   
30486  BIGHIT|"빅히트"|"방탄소년단"|"BTS"|"BANGTAN"|"방탄"|"FAK...  73463137  4924056   
30291  BIGHIT|"빅히트"|"방탄소년단"|"BTS"|"BANGTAN"|"방탄"|"FAK...  65396157  4750254   

       dislikes  comment_count  \
30844    172570        1142274   
30583    165854        1114800   
30352    156026        1084435   
30486    156026        1084435   
30291    141966        1040912   

                                       thumbnail_link  comments_disabled  \
30844  https://i.ytimg.com/vi/7C2z4GqqS5E/default.jpg              False   
30583  https://i.ytimg.com/vi/7C2z4GqqS5E/default.jpg              False   
30352  https://i.ytimg.com/vi/7C2z4GqqS5E/default.jpg              False   
30486  https://i.ytimg.com/vi/7C2z4GqqS5E/default.jpg              False   
30291  https://i.ytimg.com/vi/7C2z4GqqS5E/default.jpg              False   

       ratings_disabled  video_error_or_removed  \
30844             False                   False   
30583             False                   False   
30352             False                   False   
30486             False                   False   
30291             False                   False   

                                             description  
30844  BTS (방탄소년단) 'FAKE LOVE' Official MVDirector : ...  
30583  BTS (방탄소년단) 'FAKE LOVE' Official MVDirector : ...  
30352  BTS (방탄소년단) 'FAKE LOVE' Official MVDirector : ...  
30486  BTS (방탄소년단) 'FAKE LOVE' Official MVDirector : ...  
30291  BTS (방탄소년단) 'FAKE LOVE' Official MVDirector : ...

In [21]:
KR_sort.shape

(34567, 16)

In [23]:
KR_sort['ranking_group']=sorted(list(range(0,13))*2659)
print(KR_sort.groupby('ranking_group').views.mean())

ranking_group
0     3.022740e+06
1     6.317081e+05
2     4.047206e+05
3     3.065153e+05
4     2.385075e+05
5     1.945805e+05
6     1.559135e+05
7     1.379263e+05
8     1.163342e+05
9     8.986775e+04
10    6.553231e+04
11    5.083189e+04
12    1.091367e+05
Name: views, dtype: float64


댓글수로 정렬하여 13개의 그룹으로 나누어 본 결과, 확실히 각 그룹별 조회수 평균이 댓글수와 비슷한 경향을 띠는 것으로 보인다.

### 3. like 와 dislike (호감/비호감) 비율 구한 뒤, 어떤 카테고리인지 파악하기

In [1]:
# 패키지 불러오기
from pandas import *
import pandas as pd

In [2]:
# 현재 위치 확인 
import os 
os.getcwd() # 현재 디렉토리 위치 확인 

'C:\\Users\\Chanmi Yoo\\Desktop\\ESAA\\191011 ESAA 팀'

In [3]:
# 데이터 불러오기 
input_fd = open('KRvideos.csv',errors = 'backslashreplace',encoding='utf-8')
KRvideos = pd.read_csv(input_fd)

KRvideos.head()

video_id trending_date                       title channel_title  \
0  RxGQe4EeEpA      17.14.11          좋아 by 민서_윤종신_좋니 답가        라푸마코리아   
1  hH7wVE8OlQ0      17.14.11            JSA 귀순 북한군 총격 부상        Edward   
2  9V8bnWUmE9U      17.14.11  나몰라패밀리 운동화 영상 2탄 (빼빼로데이버젼)     나몰라패밀리 핫쇼   
3  0_8py-t5R80      17.14.11  이명박 출국 현장, 놓치면 안되는 장면         미디어몽구   
4  bk55RbxiQdI      17.14.11     김장겸은 물러갔다 MBC 노조 환호와 눈물        NocutV   

   category_id              publish_time  \
0           22  2017-11-13T07:07:36.000Z   
1           25  2017-11-13T10:59:16.000Z   
2           22  2017-11-11T07:16:08.000Z   
3           25  2017-11-12T11:19:52.000Z   
4           25  2017-11-13T11:08:59.000Z   

                                                tags   views  likes  dislikes  \
0                      라푸마|"윤종신"|"좋니"|"좋아"|"샬레"|"민서"  156130   1422        40   
1        JSA|"귀순"|"북한군"|"총격"|"부상"|"JSA 귀순 북한군 총격 부상"   76533    211        28   
2                아디다스|"빼빼로"|"핫쇼"|"나몰라패밀리"|"대학로"|"공연"  421409   5112       166   
3  이명박|"이명박 출국금지"|"이명박 출국현장"|"이명박 바레인"|"이명박 대통령"|...  222850   2093       173   
4          nocutV|"노컷V"|"CBS"|"mbc"|"김장겸"|"해임"|"허일후"   84466   1094       109   

   comment_count                                  thumbnail_link  \
0            272  https://i.ytimg.com/vi/RxGQe4EeEpA/default.jpg   
1            113  https://i.ytimg.com/vi/hH7wVE8OlQ0/default.jpg   
2            459  https://i.ytimg.com/vi/9V8bnWUmE9U/default.jpg   
3           1219  https://i.ytimg.com/vi/0_8py-t5R80/default.jpg   
4            450  https://i.ytimg.com/vi/bk55RbxiQdI/default.jpg   

   comments_disabled  ratings_disabled  video_error_or_removed  \
0              False             False                   False   
1              False             False                   False   
2              False             False                   False   
3              False             False                   False   
4              False             False                   False   

                                         description  
0  윤종신 '좋니'의 답가 '좋아' 최초 공개!\n그 여자의 이야기를 지금 만나보세요....  
1  [채널A단독]北 병사 현재 '의식불명'… 혈압 떨어지는 중 \n[채널A단독]우리측 ...  
2                                    퍼가실때 꼭 출처 부탁드려요  
3                                다스는 누구겁니까 듣고 난 후 표정  
4  김장겸 MBC 사장이 결국 해임됐다.\n\nMBC 대주주인 방송문화진흥회는 13일 ...

In [4]:
# 카데고리 설명 json 파일 불러오기 
import json
from pprint import pprint

with open('KR_category_id.json') as data_file:    
    data = json.load(data_file)
    
pprint(data)

{'etag': '"XI7nbFXulYBIpL0ayR_gDh3eu1k/1v2mrzYSYG6onNLt2qTj13hkQZk"',
 'items': [{'etag': '"XI7nbFXulYBIpL0ayR_gDh3eu1k/Xy1mB4_yLrHy_BmKmPBggty2mZQ"',
            'id': '1',
            'kind': 'youtube#videoCategory',
            'snippet': {'assignable': True,
                        'channelId': 'UCBR8-60-B28hp2BmDPdntcQ',
                        'title': 'Film & Animation'}},
           {'etag': '"XI7nbFXulYBIpL0ayR_gDh3eu1k/UZ1oLIIz2dxIhO45ZTFR3a3NyTA"',
            'id': '2',
            'kind': 'youtube#videoCategory',
            'snippet': {'assignable': True,
                        'channelId': 'UCBR8-60-B28hp2BmDPdntcQ',
                        'title': 'Autos & Vehicles'}},
           {'etag': '"XI7nbFXulYBIpL0ayR_gDh3eu1k/nqRIq97-xe5XRZTxbknKFVe5Lmg"',
            'id': '10',
            'kind': 'youtube#videoCategory',
            'snippet': {'assignable': True,
                        'channelId': 'UCBR8-60-B28hp2BmDPdntcQ',
                        'title': 'Music'}},
 

In [14]:
# 카테고리 설명 데이터프레임 만들기 
ID = []
title = []

for i in range(len(data["items"])):
    ID.append(data["items"][i]["id"])
    title.append(data["items"][i]["snippet"]["title"])

category_desc = pd.DataFrame({'category_id':ID, 
                             'category': title})
category_desc

category_id              category
0            1      Film & Animation
1            2      Autos & Vehicles
2           10                 Music
3           15        Pets & Animals
4           17                Sports
5           18          Short Movies
6           19       Travel & Events
7           20                Gaming
8           21         Videoblogging
9           22        People & Blogs
10          23                Comedy
11          24         Entertainment
12          25       News & Politics
13          26         Howto & Style
14          27             Education
15          28  Science & Technology
16          30                Movies
17          31       Anime/Animation
18          32      Action/Adventure
19          33              Classics
20          34                Comedy
21          35           Documentary
22          36                 Drama
23          37                Family
24          38               Foreign
25          39                Horror
26          40        Sci-Fi/Fantasy
27          41              Thriller
28          42                Shorts
29          43                 Shows
30          44              Trailers

31가지 카테고리가 있다.

In [16]:
# 문자열 변환 
category_desc["category_id"] = category_desc["category_id"].apply(str)
KRvideos["category_id"] = KRvideos["category_id"].apply(str)

# 기존 KR 데이터프레임에 카테고리 설명 데이터프레임을 합친 뒤, KR에 저장하기 
KR = pd.merge(KRvideos, category_desc, on="category_id", how='left')

video_id trending_date               title channel_title category_id  \
0  RxGQe4EeEpA      17.14.11  좋아 by 민서_윤종신_좋니 답가        라푸마코리아          22   
1  hH7wVE8OlQ0      17.14.11    JSA 귀순 북한군 총격 부상        Edward          25   

               publish_time                                         tags  \
0  2017-11-13T07:07:36.000Z                라푸마|"윤종신"|"좋니"|"좋아"|"샬레"|"민서"   
1  2017-11-13T10:59:16.000Z  JSA|"귀순"|"북한군"|"총격"|"부상"|"JSA 귀순 북한군 총격 부상"   

    views  likes  dislikes  comment_count  \
0  156130   1422        40            272   
1   76533    211        28            113   

                                   thumbnail_link  comments_disabled  \
0  https://i.ytimg.com/vi/RxGQe4EeEpA/default.jpg              False   
1  https://i.ytimg.com/vi/hH7wVE8OlQ0/default.jpg              False   

   ratings_disabled  video_error_or_removed  \
0             False                   False   
1             False                   False   

                                         description         category  
0  윤종신 '좋니'의 답가 '좋아' 최초 공개!\n그 여자의 이야기를 지금 만나보세요....   People & Blogs  
1  [채널A단독]北 병사 현재 '의식불명'… 혈압 떨어지는 중 \n[채널A단독]우리측 ...  News & Politics

### [좋아요(likes) 상위 25%의 카테고리 분포]

In [22]:
# 좋아요 내림차순으로 정렬 후 KR_likes에 저장 
KR_likes = KR.sort_values(by=['likes'], ascending=False)

# 좋아요 상위 25% 추출 후 KR_25에 저장 
KR_likes25 = KR_likes.iloc[1:round(len(KR_likes)*0.25)]
KR_likes25.head(3)

video_id trending_date                                title  \
30583  7C2z4GqqS5E      18.23.05  BTS (방탄소년단) 'FAKE LOVE' Official MV   
30352  7C2z4GqqS5E      18.22.05  BTS (방탄소년단) 'FAKE LOVE' Official MV   
30486  7C2z4GqqS5E      18.22.05  BTS (방탄소년단) 'FAKE LOVE' Official MV   

      channel_title category_id              publish_time  \
30583       ibighit          10  2018-05-18T09:00:02.000Z   
30352       ibighit          10  2018-05-18T09:00:02.000Z   
30486       ibighit          10  2018-05-18T09:00:02.000Z   

                                                    tags     views    likes  \
30583  BIGHIT|"빅히트"|"방탄소년단"|"BTS"|"BANGTAN"|"방탄"|"FAK...  80738011  5053338   
30352  BIGHIT|"빅히트"|"방탄소년단"|"BTS"|"BANGTAN"|"방탄"|"FAK...  73463137  4924056   
30486  BIGHIT|"빅히트"|"방탄소년단"|"BTS"|"BANGTAN"|"방탄"|"FAK...  73463137  4924056   

       dislikes  comment_count  \
30583    165854        1114800   
30352    156026        1084435   
30486    156026        1084435   

                                       thumbnail_link  comments_disabled  \
30583  https://i.ytimg.com/vi/7C2z4GqqS5E/default.jpg              False   
30352  https://i.ytimg.com/vi/7C2z4GqqS5E/default.jpg              False   
30486  https://i.ytimg.com/vi/7C2z4GqqS5E/default.jpg              False   

       ratings_disabled  video_error_or_removed  \
30583             False                   False   
30352             False                   False   
30486             False                   False   

                                             description category  
30583  BTS (방탄소년단) 'FAKE LOVE' Official MVDirector : ...    Music  
30352  BTS (방탄소년단) 'FAKE LOVE' Official MVDirector : ...    Music  
30486  BTS (방탄소년단) 'FAKE LOVE' Official MVDirector : ...    Music

In [23]:
# 좋아요 상위 25%의 카테고리 분포 살펴보기 
KR_likes25['category'].astype(object)
KR_likes25.category.value_counts()

Entertainment           2377
Music                   1153
People & Blogs          1130
Comedy                  1031
News & Politics          976
Film & Animation         569
Gaming                   478
Pets & Animals           395
Howto & Style            200
Sports                   175
Education                 84
Science & Technology      47
Travel & Events           15
Autos & Vehicles           6
Name: category, dtype: int64

#### 좋아요 수가 많은(상위 25% 수준) 유투브 동영상의 카데고리 분포를 살펴본 결과, Entertainment가 2377개로 가장 많았고, Music과 People&Blogs가 각각 1153개와 1130개로 그 뒤를 이었으며, Comedy도 1031개로 많았다. 

In [24]:
round(KR_likes25.category.value_counts()/len(KR_likes25)*100, 2)

Entertainment           27.51
Music                   13.34
People & Blogs          13.08
Comedy                  11.93
News & Politics         11.29
Film & Animation         6.58
Gaming                   5.53
Pets & Animals           4.57
Howto & Style            2.31
Sports                   2.03
Education                0.97
Science & Technology     0.54
Travel & Events          0.17
Autos & Vehicles         0.07
Name: category, dtype: float64

#### Entertainment 27.51%, Music 13.34%, People & Blogs 13.08%, Comedy 11.93%, News & Politics 11.29%로 집계되었다.

#### 즉, 한국인에게 좋아요를 많이 받은 유투브 영상은 주로 연예, 음악, v-log 콘텐츠인 것으로 드러났으며, 교육, 과학기술, 여행, 자동차 관련 콘텐츠는 좋아요 수가 많지 않았다.

### [싫어요(dislikes) 상위 25%의 카테고리 분포]

In [28]:
# 같은 방식으로 싫어요 상위 25%의 카테고리 분포를 살펴보자

KR_dislikes = KR.sort_values(by=['dislikes'], axis=0, ascending=False)
KR_dislikes25 = KR_dislikes.iloc[1:round(len(KR_dislikes)*0.25)]
KR_dislikes25['category'].astype(object)
KR_dislikes25.category.value_counts()

Entertainment           2548
News & Politics         1811
People & Blogs          1327
Music                    759
Comedy                   714
Film & Animation         547
Gaming                   388
Sports                   249
Howto & Style             76
Education                 56
Pets & Animals            52
Science & Technology      46
Travel & Events           20
Autos & Vehicles          14
Shows                      8
Name: category, dtype: int64

#### 싫어요 수가 많은 카테고리는 Entertainment(2548개), News & Politics(1811개), People & Blogs(1327개), Music(759개) 순인 것으로 나타났다. 


In [29]:
round(KR_dislikes25.category.value_counts()/len(KR_dislikes25)*100, 2)

Entertainment           29.49
News & Politics         20.96
People & Blogs          15.36
Music                    8.78
Comedy                   8.26
Film & Animation         6.33
Gaming                   4.49
Sports                   2.88
Howto & Style            0.88
Education                0.65
Pets & Animals           0.60
Science & Technology     0.53
Travel & Events          0.23
Autos & Vehicles         0.16
Shows                    0.09
Name: category, dtype: float64

#### 비율을 살펴보면, Entertainment가 29.49%로 가장 많았고, News & Politics(20.96%)과 People & Blogs(15.36%)로 그 뒤를 이었으며, Howto & Style, Education, Pets & Animals, Science & Technology, Travel & Events, Autos & Vehicles, Shows는 싫어요를 많이 받지 않았다. 

In [32]:
KR_likes25["title"].head(20)

30583                  BTS (방탄소년단) 'FAKE LOVE' Official MV
30352                  BTS (방탄소년단) 'FAKE LOVE' Official MV
30486                  BTS (방탄소년단) 'FAKE LOVE' Official MV
30291                  BTS (방탄소년단) 'FAKE LOVE' Official MV
30171                  BTS (방탄소년단) 'FAKE LOVE' Official MV
30069                  BTS (방탄소년단) 'FAKE LOVE' Official MV
29969                  BTS (방탄소년단) 'FAKE LOVE' Official MV
4856     YouTube Rewind: The Shape of 2017 | #YouTubeRe...
4678     YouTube Rewind: The Shape of 2017 | #YouTubeRe...
2788     BTS (방탄소년단) 'MIC Drop (Steve Aoki Remix)' Offi...
4018     Marvel Studios' Avengers: Infinity War Officia...
3750     Marvel Studios' Avengers: Infinity War Officia...
2563     BTS (방탄소년단) 'MIC Drop (Steve Aoki Remix)' Offi...
28605    BTS (방탄소년단) LOVE YOURSELF 轉 Tear 'Singularity'...
3544     Marvel Studios' Avengers: Infinity War Officia...
28371    BTS (방탄소년단) LOVE YOURSELF 轉 Tear 'Singularity'...
28554    Childish Gambino - This Is America (Official V.

In [33]:
KR_dislikes25["title"].head(20)

4678     YouTube Rewind: The Shape of 2017 | #YouTubeRe...
4453     YouTube Rewind: The Shape of 2017 | #YouTubeRe...
4244     YouTube Rewind: The Shape of 2017 | #YouTubeRe...
30844                  BTS (방탄소년단) 'FAKE LOVE' Official MV
30583                  BTS (방탄소년단) 'FAKE LOVE' Official MV
30486                  BTS (방탄소년단) 'FAKE LOVE' Official MV
30352                  BTS (방탄소년단) 'FAKE LOVE' Official MV
30171                  BTS (방탄소년단) 'FAKE LOVE' Official MV
30291                  BTS (방탄소년단) 'FAKE LOVE' Official MV
28554    Childish Gambino - This Is America (Official V...
29969                  BTS (방탄소년단) 'FAKE LOVE' Official MV
30069                  BTS (방탄소년단) 'FAKE LOVE' Official MV
28803    CHẠY NGAY ĐI | RUN NOW | SƠN TÙNG M-TP | Offic...
28313    Childish Gambino - This Is America (Official V...
24314                              TWICE What is Love? M/V
24088                              TWICE What is Love? M/V
5991                                TWICE Heart Shaker M

In [52]:
total_25 = pd.concat([KR_likes25, KR_dislikes25])
total_25["duplicated"] = total_25.duplicated()

In [68]:
# 좋아요 상위 25% 영상 개수
print(KR_likes25.shape[0])
# 싫어요 상위 25% 영상 개수
print(KR_dislikes25.shape[0])
# 겹치는 영상 개수 
print(len(total_25[total_25["duplicated"]== True]))

8641
8641
5943


In [71]:
8641-5943

2698

#### 상위 25%에 해당하는 영상의 개수는 각각 8641개이며, 그 중 5943개의 영상이 겹치고 중복되지 않는 영상의 개수는 각각 2698개이다. 따라서 겹치는 영상이 많기 때문에 좋아요 카테고리와 싫어요 카테고리의 순위가 일치한다고 볼 수 있다. 
#### 좋아요 최상위권과 싫어요 최상위권은 다소 중복되는 경향이 있었는데, 제목을 살펴보면 대부분 BTS(방탄소년단)의 공식 뮤직비디오였다. 이러한 현상은 BTS 유행에 따른 호기심과 대중의 적극적 평가 등이 영향을 미쳤을 것으로 예상된다. 

#### <결론>

#### 한국 Youtube 동영상 중에서 가장 좋아요 혹은 싫어요가 많은 카테고리는 엔터테인먼트였으며, 좋아요가 많은 카테고리 순위와 싫어요가 많은 카테고리 순위가 유사했다. 이는 좋아요 순위가 높은 영상과 싫어요 순위가 높은 영상이 겹치기 때문인 것으로 보인다. Entertainment와 People&Blogs 콘텐츠에 대해 반응이 활발하게 일어나며, 뉴스나 정치적 사안을 다룬 영상에 대해서는 싫어요 수가 많다는 것을 알 수 있었다.